packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -27,8 +27,10 @@ function BlockListAppender( {
 		return null;
 	}
 
-	// A render prop has been provided, use it to render the appender.
-	if ( CustomAppender ) {
+	// A render prop has been provided
+	// and it is not a component that return `null`
+	// then use it to render the appender.
+	if ( CustomAppender && ! isNull( CustomAppender ) ) {
```

Comment:
I guess `isNull` is not correct here? We need to check that it's a function that returns a falsy value right? A function is not null?

Reply:
A simpler approach might be to do `CustomAppender === false` ? what do you think?

Reply:
As in 

`if ( CustomAppender && isFunction( CustomAppender ) && ! isNull( CustomAppender() ) ) {`?

Reply:
yeah but the second approach seems simpler. It ensures that if you do `renderAppender={ false }` there's no appender rendered? (I'm ok with both you choose :P )